8 update code to handle new data format

JimBundy opened the pull request:
Fixed a couple issues but now all ABR data (including Ivy/Andrew and new data) should work. If folders are formatted correctly, the plot_previous_peaks function should work, allowing previous picked peaks to be displayed for viewing only. Also makes it so saving doesn't overwrite old files.

JimBundy committed:
Added baseline functionality to new data

Yes

sivaprakasaman committed:
Merge branch '8-update-code-to-handle-new-data-format' into New-ABR-Data-Fix

sivaprakasaman committed:
Merge pull request #9 from HeinzLabPurdue/New-ABR-Data-Fix

Adding Jim's code to issue #8

JimBundy committed:
Fix invert and file saving

Yes

JimBundy committed:
Fixed old data breaking

Made the SampleRate fix only activate for new data, where SampleRate exists as a parameter

JimBundy committed:
Click fix

Fixed click only datasets not loading

JimBundy committed:
Fix for issues with Ivy/Andrew data

Fixed issue caused by sampling rate

JimBundy committed:
Enabled load_previous_peaks again

Load_previous_peaks works ONLY when folders are formatted right

JimBundy committed:
Save_file fix

Allows files to be saved without overwriting previous data

sivaprakasaman reviewed (changes requested):
Seems good, just check Windows compatibility and confirm test files work! Thanks!

Commit:
Added a note to change p0001_calib_raw files to p0001_calib

sivaprakasaman commented:
GREAT WORK, JIM! Seems to be running ok...now confirmed on all 3 operating systems! I'll use this version to verify Ivy's peak picking is showing up (I think one chin may have been over-written, but I can now identify those peaks!). Once the data has been checked/processed, I'll approved this merge.

sivaprakasaman commented on ABR-master/Matlab_ABR/ABR_analysis/private/load_abr_data.m:
```diff
@@ -63,12 +66,38 @@
             freqs(1,i)=0; % means click
         end
         attn(1,i)=-x.Stimuli.atten_dB;
+        if ~isfield(x.AD_Data, 'SampleRate')
```
Nice!!! Thanks for this...now code will be backwards compatible!

sivaprakasaman commented on ABR-master/Matlab_ABR/ABR_analysis/private/load_abr_data.m:
```diff
@@ -125,7 +154,8 @@
 CalibFile  = sprintf('p%04d_calib', str2double(abr_Stimuli.cal_pic));
 command_line = sprintf('%s%s%c','[xcal]=',CalibFile,';');
-eval(command_line);
+eval(command_line); % JB: If you get an error here, make sure your calib file 
```
You may want to consider throwing an error here with a message in the CL.

sivaprakasaman commented on ABR-master/Matlab_ABR/ABR_analysis/private/zzz5.m:
```diff
@@ -3,12 +3,11 @@
 
 function zzz5
```
Eventually we should rename all these "zzz" functions...bad naming convention

JimBundy committed:
p0001_calib crash fix

Now throws a descriptive warning but *probably* doesn't crash if you have p0001_calib_raw instead of p0001_calib

sivaprakasaman committed:
updating this branch with various changes from Jim PLUS this is running entirely on mat files.

sivaprakasaman committed:
Merge pull request #13 from sivaprakasaman/8-update-code-to-handle-new-data-format

updating this branch with various changes from Jim PLUS this is runni…
